File: iidaPro-2015/SearchDetailViewController.m
//
//  SearchDetailViewController.m
//  iidaPro-2015
//
//  Created by akaimo on 11/29/15.
//  Copyright © 2015 akaimo. All rights reserved.
//

#import "SearchDetailViewController.h"

@interface SearchDetailViewController ()
@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
@property (weak, nonatomic) IBOutlet UIView *genreView;
@property (weak, nonatomic) IBOutlet UILabel *genreLabel;
@property (weak, nonatomic) IBOutlet UIImageView *genreImageView1;
@property (weak, nonatomic) IBOutlet UIImageView *genreImageView2;
@property (weak, nonatomic) IBOutlet UIImageView *genreImageView3;
@property (weak, nonatomic) IBOutlet UITextView *detailTextView;

@end

@implementation SearchDetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"詳細";
    self.view.layer.contents = (id)[UIImage imageNamed:@"Base"].CGImage;
    _genreView.layer.cornerRadius = 10.0;
    _genreView.layer.masksToBounds = YES;
    
    [self setupData];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)setupData {
    _titleLabel.text = [_trashData valueForKey:@"title"];
    // TODO: 実際のデータに置き換える
    _genreImageView1.image = [self setGenreImage:[_trashData valueForKey:@"category"]];
    _detailTextView.text = @"分解できるなら... \n hogehogehogehogehogehoge";
    _detailTextView.textColor = [UIColor whiteColor];
}

- (UIImage *)setGenreImage:(NSString *)genre {
    UIImage *img;
    if ([genre  isEqual: @"普通ごみ"]) {
        img = [UIImage imageNamed:@"S_Normal"];
    } else if ([genre  isEqual: @"ミックスペーパー"]) {
        img = [UIImage imageNamed:@"S_Mixed"];
    } else if ([genre  isEqual: @"プラスチック製容器包装"]) {
        img = [UIImage imageNamed:@"S_plastic"];
    } else if ([genre  isEqual: @"小物金属"]) {
        img = [UIImage imageNamed:@"S_Metal"];
    } else if ([genre  isEqual: @"使用済み乾電池"]) {
        img = [UIImage imageNamed:@"S_battery"];
    } else if ([genre  isEqual: @"空き缶・ペットボトル"]) {
        img = [UIImage imageNamed:@"S_Can"];
    } else if ([genre  isEqual: @"粗大ごみ"]) {
        img = [UIImage imageNamed:@"S_BigRefure"];
    }
    
    return img;
}

@end

File: iidaPro-2015/SearchDetailViewController.swift
//
//  SearchDetailViewController.swift
//  iidaPro-2015
//
//  Created by akaimo on 12/26/15.
//  Copyright © 2015 akaimo. All rights reserved.
//

import UIKit

class SearchDetailViewController: UIViewController {
    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var genreView: UIView!
    @IBOutlet weak var genreImageView1: UIImageView!
    @IBOutlet weak var genreImageView2: UIImageView!
    @IBOutlet weak var genreImageView3: UIImageView!
    @IBOutlet weak var detailTextView: UITextView!
    
    var trashData: AnyObject?

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.title = "詳細"
        self.setupView()
    }

    override func viewDidLayoutSubviews() {
        self.detailTextView.setContentOffset(CGPoint.zero, animated: false)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    
    private func setupView() {
        self.view.layer.contents = UIImage(named: "Base")?.CGImage
        
        self.genreView.layer.cornerRadius = 10.0
        self.genreView.layer.masksToBounds = true
        
        guard let data = self.trashData as? TrashCategory else { return }
        self.titleLabel.text = data["title"] as? String
        self.detailTextView.text = data["info"] as? String
        self.detailTextView.textColor = UIColor.whiteColor()
        self.detailTextView.font = UIFont.systemFontOfSize(16)
        
        guard let trashStr = data["allCategory"] as? String else { return }
        let array = trashStr.characters.split("|").map { String($0) }
        let viewArray = [self.genreImageView1, self.genreImageView2, self.genreImageView3]
        for num in 0..<array.count {
            viewArray[num].image = self.selectGenreImage(array[num])
        }
    }
    
    private func selectGenreImage(genre: String) -> UIImage? {
        var img: UIImage?
        
        switch genre {
        case "普通ごみ": img = UIImage(named: "S_Normal")
        case "ミックスペーパー": img = UIImage(named: "S_Mixed")
        case "プラスチック製容器包装": img = UIImage(named: "S_plastic")
        case "小物金属": img = UIImage(named: "S_Metal")
        case "使用済み乾電池": img = UIImage(named: "S_battery")
        case "空き缶・ペットボトル": img = UIImage(named: "C_Can")
        case "空きびん": img = UIImage(named: "C_Can")
        case "粗大ごみ": img = UIImage(named: "S_BigRefuse")
        case "収集しない": img = UIImage(named: "S_No")
        default: break
        }
        
        return img
    }
    
}
